### ISSUE TITLE
Backend swallows exception during spatial dataset creation

### ISSUE THREAD

**Issue by ansoncfit** [conveyal/r5#727]
Backend log shows:
```
01:49:12.449 [pool-3-thread-2] INFO  c.c.a.c.OpportunityDatasetController - Detected opportunity dataset stored as ESRI shapefile.
01:49:25.732 [pool-3-thread-2] ERROR c.c.a.components.TaskScheduler - Background execution of class com.conveyal.analysis.controllers.OpportunityDatasetController$$Lambda$464/0x000000080069d040 caused exception: java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space
        at com.conveyal.r5.analyst.Grid.<init>(Grid.java:147)
        at com.conveyal.r5.analyst.Grid.lambda$fromShapefile$0(Grid.java:725)
```

But UI still shows the dataset as PROCESSING.

**Comment by abyrd** [comment]
Migrating this to use the same background task mechanism as bundle processing will provide a uniform channel to report errors. I would expect to do this by the next release. This error is problematic though - we don’t want out of memory situations at all. Maybe need to check the size or other aspects of inputs before processing if this step has the potential to exhaust memory.

**Comment by ansoncfit** [comment]
Using
> the same background task mechanism as bundle processing

should be implemented as part of the next round of #732, or a subsequent PR moving opportunity dataset creation to the two-step upload then convert workflow

**Comment by ansoncfit** [comment]
Original issue fixed; remaining part opened as https://github.com/conveyal/r5/issues/761
